rclpy/src/rclpy/_rclpy.c

```diff
@@ -1345,6 +1345,20 @@ rclpy_destroy_node_entity(PyObject * Py_UNUSED(self), PyObject * args)
   }
 
   rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, NULL);
+
+  if (!node) {
+    PyErr_SetString(PyExc_ValueError,
+      "rcl node is null");
+    return NULL;
+  }
+
+  if (!rcl_node_is_valid(node)) {
+    PyErr_SetString(PyExc_ValueError,
+      "rcl node is invalid");
+    rcl_reset_error();
+    return NULL;
+  }
```

Comment:
Or at least they should be checked as of your pr https://github.com/ros2/rcl/pull/148.